perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -695,8 +723,64 @@ def ListConsistencyBenchmark(results, metadata, vm, command_builder,
                                    metadata)
 
 
+def LoadWorkerOutput(output):
+  """Load output from worker processes to our internal format.
+
+  Args:
+    output: list of strings. The stdouts of all worker processes.
+
+  Returns:
+    A tuple of start_time, latency, size. Each of these is a list of
+    numpy arrays, one array per worker process. The numpy arrays have
+    one entry per object sent by the worker, and store the
+    corresponding value for that object transfer. start_time and
+    latency are float64s, storing POSIX timestamps and durations in
+    seconds. size stores int64s, holding object size in
+    bytes.
+
+  Raises:
+    AssertionError, if an individual worker's input includes
+    overlapping operations, or operations that don't move forward in
+    time, or if the input list isn't in stream number order.
+  """
+
+  start_times = []
+  latencies = []
+  sizes = []
+
+  prev_stream_num = None
+  for worker_out in output:
+    json_out = json.loads(worker_out)
+    num_records = len(json_out)
+
+    assert (not prev_stream_num or
+            json_out[0]['stream_num'] == prev_stream_num + 1)
+    prev_stream_num = prev_stream_num + 1 if prev_stream_num else 0
+
+    start_time = np.zeros([num_records], dtype=np.float64)
+    latency = np.zeros([num_records], dtype=np.float64)
+    size = np.zeros([num_records], dtype=np.int64)
+
+    prev_start = None
+    prev_latency = None
+    for i in xrange(num_records):
```

Comment:
@hildrum No reason, your version of the loop is better. I'd like to push a new commit anyway to fix some confusing variable names, so I will also make the change then.